state/cluster/badger/mutator.go

```diff
@@ -121,6 +130,9 @@ func (m *Mutator) Extend(block *cluster.Block) error {
 		// do this by tracing back until we see a parent block that is the
 		// latest finalized block, or reach height below the finalized boundary
 
+		m.state.tracer.FinishSpan(blockID, trace.COLClusterStateMutatorExtendSetup)
```

Comment:
What would happen if the `FinishSpan` is not called when exits earlier?

There is a `Run` pattern that might fits here:

Instead of wrapping the section with StartSpan and FinishSpan, which might miss the call to FinishSpan when exited earlier:
```
trace.StartSpan(id, taskID)

err := doSubTask1()
if err != nil {
  return err
}
err := doSubTask2()
if err != nil {
  return err
}

trace.FinishSpan(id, taskID)
```

We could do :

```
tracker.RunTrace(id, taskID, func() {
  err := doSubTask1()
  if err != nil {
    return err
  }
  err := doSubTask2()
  if err != nil {
    return err
  }
})
```

And `RunTrace` with wrap with the tracing:

```
func (t *Tracer) RunTrace(id, taskID, f) {
  t.StartSpan(id, taskID)
  defer t.FinishSpan(id, taskID)

  f()
}
```

Reply:
Good point. `I think the `RunTrace` might be tricky because the `f` function needs to impact control flow at the top level of the function being traced, so we would need to return and check the error every time. Also I think quite a few variables are defined in one span and used later on etc. 

`FinishTrace` is a noop when called multiple times, so we could just `defer` the `FinishTrace` always, even if we also manually finish the span at a later time.

Reply:
Example: https://github.com/onflow/flow-go/pull/42/commits/f66c8e6734f5bed88d799862431ad14fc05cf959

Reply:
Ok, that works too